goosebit/db/models.py

```diff
@@ -130,18 +129,3 @@ async def latest(cls, device: Device) -> Self | None:
             key=lambda x: semver.Version.parse(x.version),
             reverse=True,
         )[0]
-
```

Comment:
Is it a good idea to move such logic into the Pydantic schema? The way I saw the schemas until now was that they represent our DTO (data transfer models). As such, they only care about serialisation/deserialisation and validation.

Depending on our namespaces (v1 vs bff) we will also have different schemas/DTOs for the same model (like Software in this case).

Reply:
The goal here would be to allow a plugin to extend that schema for it's own purposes, and still be able to conform to the internal stuff.  EG for a plugin we want to build, we would need to be able to push alternate one-off software updates to a device.  This means that passing around the `Software` model internally isn't ideal, since creating a new DB item for each devices' one-off updates will clutter up the UI, mess up the update handler, etc.

Maybe in this case there is some way to build a "one-shot" updater, but I still think generalizing this as much as possible would be nice.

Reply:
I have kept it in the original `Software` model as well now, but never hurts to include it here as well.